### PR TITLE
Created new API endpoint to fetch all available product filters and added authentication classes for DRF

### PR DESCRIPTION
**Pull request by msaurabh50702**

### CODE REVIEW

File: roopsangamnx_backend/inventory/serializers.py [hunk 1]
```diff
@@ -6,28 +6,29 @@
 class SubcategorySerializer(serializers.ModelSerializer):
     class Meta:
         model = SubCategory
-        fields = ['name']
+        fields = "__all__"
```
Comment (saichothave) [inline roopsangamnx_backend/inventory/serializers.py:9]: do we required created and updated at fields in response, its making crowded response

File: roopsangamnx_backend/inventory/views.py [hunk 2]
```diff
@@ -118,4 +119,21 @@ def get_or_create_product_color(self, color_data):
         color, created = ProductColor.objects.get_or_create(color=color_name)
         return color
 
+class FilterView(APIView):
```
Comment (saichothave) [inline roopsangamnx_backend/inventory/views.py:122]: we already had filters for each endpoint of products, categories, subcategories, now this in another api which is also applying filters?

**Review by saichothave — APPROVED** [review]

File: roopsangamnx_backend/inventory/serializers.py (hunk 1 above)
Comment (msaurabh50702) [reply roopsangamnx_backend/inventory/serializers.py:9]: Created & modified is not required, name & id is sufficient

File: roopsangamnx_backend/inventory/views.py (hunk 2 above)
Comment (msaurabh50702) [reply roopsangamnx_backend/inventory/views.py:122]: Created seperate endpoint to get all available details for categories, subcategories, colors and size 

Will send request to available filters with this details so product gets filtered accordingly 

This endpoint will handle only GET method to return above fields in single api call

**Pull request merged**